local-volume/provisioner/pkg/common/common.go

```diff
@@ -182,6 +203,18 @@ func ConfigMapDataToVolumeConfig(data map[string]string, provisionerConfig *Prov
 	if err := yaml.Unmarshal([]byte(rawYaml), provisionerConfig); err != nil {
 		return fmt.Errorf("fail to Unmarshal yaml due to: %#v", err)
 	}
+	for class, config := range provisionerConfig.StorageClassConfig {
+		if config.BlockCleanerCommand == nil {
+			// supply a default block cleaner command.
+			config.BlockCleanerCommand = []string{DefaultBlockCleanerCommand}
+		} else {
+			// Validate that array is non empty
+			if len(config.BlockCleanerCommand) < 1 {
+				return fmt.Errorf("Invalid empty block cleaner command for class %v", class)
+			}
+		}
+		provisionerConfig.StorageClassConfig[class] = config
```

Comment:
config is a local variable?

Reply:
Yes, but the value is being saved back into the map. Go doesn't allow changing a struct member when the struct is in a map, one has the set the entire struct.